### DISCUSSION TITLE
Switched to config from PR 200, now updates are nonstop

### DISCUSSION THREAD

**Discussion by jcam** [emporia-vue-local/esphome#202] [Q&A] [answered]
I rebuilt one of my nodes following the updates from PR200 using a 5 second time window, and now I see on_update is triggering all the updates as fast as it possibly can (though the data only changes every 5 seconds, the log shows it is spamming my HA box/network with useless data)

Anyone else experiencing this?

**Comment by kahrendt** [comment, marked as answer]
When you say it is spamming HA, do you mean log messages? My understanding (which could be wrong) is that log messages are only sent if actively connected through the ESPHome command line, though I believe if you use MQTT then it always sends the logs to a specific MQTT topic. Regardless, these can be removed. You can decrease the amount of log messages by lowering the logger level to WARN (or lower). This, in general, improves performance quite a bit. 

Try adding this to your config

```
logger:
  baud_rate: 0   # don't send log messages over UART interface
  level: WARN

```

If that works well for you and others, then I think we should add it to the base configuration in the repository. The component is quite stable on my end, so I don't think seeing the full debug messages is particularly useful.

**Reply by jcam** [reply]
Mmmmm good points! I haven't actually gone to check what is actually being sent to the HA API endpoint, I did make the assumption based on what was hitting the esphome CLI logger output that all of that was going to HA.
I'm turning off the MQTT logger (I use the MQTT topic to get overall house usage and solar production into PVOutput.org and my other solar production tracking systems) and turning off debug logging overall.
I wonder if there's a simple way to watch everything that hits the HA endpoint?

I was making assumptions as well...
I thought this would result in 2 sensors, one that has an average and the other that updates constantly, but I don't see that in the output so perhaps not?
sensor:
  - platform: emporia_vue
    ct_clamps:
      - { phase_id: phase_b, input:  "1", power: { id:  cir2, filters: [ *pos, multiply: 2 ] } }
  - { platform: copy, name:  "Lower BR Mini", source_id:  cir2, filters: *throttle_avg }

**Reply by kahrendt** [reply]
The first sensor is internal to ESPHome. If you only give an ``id`` (in this case ``cir2``) but do not define a ``name``, then the sensor is automatically internal; i.e., it doesn't publish to HA or MQTT. It however can be referenced elsewhere, which we do with the ``copy`` sensor. Since that one has a ``name`` defined, it is by default not internal, so it will automatically publish its state to HA or MQTT.

Some access points/routers give information about how much data is sent and received for each wifi device (my Unifi APs can do this). You could run a test where you connect via the command line for logs and measure how much data is sent over a period of time. Then disconnect, and measure over the same period of time. If it is significantly less, then that means the log messages are not being sent to HA. Other than that, I am unsure of an easy way to test exactly what is sent to HA.

I actually use both MQTT and HA. I have all the MQTT data collected straight into a long-term database, whereas the HA data is only used for quick reference and the energy dashboard. You mentioned disabling the logging via MQTT, but for other people's reference, you can disable MQTT logging by adding this when you configure the base MQTT component:

```
mqtt:
  ... (usual configuration stuff for MQTT)
  log_topic:
    level: NONE
```

**Reply by jcam** [reply]
you can also set this for the same effect:
```
mqtt:
  ...
  log_topic: null
```

**Reply by flaviut** [reply]
Great that y'all figured this out! I've updated the readme in https://github.com/emporia-vue-local/esphome/commit/ded4086daa5f01e68760543fb7c677d5b3c02963 with a different configuration that ought to also fix this:

```
logger:
  logs:
    # by default, every reading will be printed to the UART, which is very slow
    # This will disable printing the readings but keep other helpful messages
    sensor: INFO
```

See also https://github.com/emporia-vue-local/esphome/pull/204
